Tests/SonarScanner.MSBuild.Common.Test/MsBuildPathSettingsTests.cs

```diff
@@ -237,12 +239,30 @@ public void GetImportBeforePaths_Windows_System_Account_Sysnative_Missing()
                 });
         }
 
+        [DataTestMethod]
+        [DataRow("/Users/runner/.local/share", DisplayName = "NET 7.0 and earlier")]
+        [DataRow("/Users/runner/Library/ApplicationSupport", DisplayName = "NET 8.0")]
+        [DataRow("/Users/runner/Something/Different", DisplayName = "Future value")]
+        public void GetImportBeforePaths_MacOSX_ReturnsBothOldAndNewLocations(string localApplicationData)
+        {
+            var paths = new[]
+            {
+                (Environment.SpecialFolder.LocalApplicationData, localApplicationData),
+                (Environment.SpecialFolder.UserProfile, "/Users/runner"),
+            };
+
+            var result = MsBuildPathSettings(paths, false, _ => true, true).GetImportBeforePaths();
+
+            result.Should().Contain(path => path.Contains(Path.Combine("local", "share")));
+            result.Should().Contain(path => path.Contains(Path.Combine("Library", "Application Support")));
```

Comment:
We do know the exact path here. I would change the assertion to
```suggestion
            result.Should().BeEquivalentTo(new[]
            {
                "/Users/runner/.local/share",
                "/Users/runner/Library/ApplicationSupport",
            });
```
If the third case fails, add a separate test case for it.

Reply:
`GetImportBeforePaths` returns more than 20 different paths, since it resolves the paths for all MSBuild versions.
I could hardcode the list of 20 paths, but then I would be testing for way more than "returns both old and new locations". For example, if a new `Microsoft\MSBuild\<version>\` is added, or an old one needs to be removed, the test would break and would need to be adapted. That seems a `Focused` issue to me.

Reply:
Do we have a test that lists all 20 paths? This is a valuable test to have.
For this test, I would say `Contain` but with constant strings instead of a lambda, which is okay.